wordpress-mobile/WordPress-Android
# Add Sentry User Data

Adds WordPress.com account userID, username, and email address to crash logs.

**To test:**
1. Smoke test – launch the app + log in to ensure that Sentry is initialized correctly and the DI stuff is working.
2. Add `require(false)` to a Kotlin file where you can trigger it (I'd suggest `PagesActivity.kt`)
3. Add `return true` to the top of `CrashLogging.shouldSendEvents` to ensure events are sent to Sentry regardless of environment.
4. Run (not debug) this branch on-device, then trigger the crash and relaunch the app. 
5. Verify that the crash shows up in Sentry with your user details associated.

**Next Steps:**
- I'd like to get the app using the Tracks version of Sentry in order to make things consistent with the other apps, but that should be a separate PR.
- If a user logs in and experiences a crash in the same session, their user data will not be sent with the crash. The Tracks version of the Sentry integration has a mechanism for handling this transition, so I'd like to do that work once.

PR submission checklist:
